### ISSUE TITLE
[MemberNotNull] annotations on setters are incorrectly ignored when property is accessed for both read and write at the same time

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#43584]
**Version Used**: 

```
commit a1b867054dff97e5e5f44c7ad742c8ffb740b2eb
Merge: 8f4fdad99ae 6bf71e1a37d
Author: msftbot[bot] <48340428+msftbot[bot]@users.noreply.github.com>
Date:   Wed Apr 22 09:50:57 2020 +0000

    Merge pull request #43554 from CyrusNajmabadi/renameOOP4

    Move the 'finding' portoin of rename out of process.
```

**Steps to Reproduce**:

Compile the following code
```cs
using System.Diagnostics.CodeAnalysis;

class C
{
  public string? Prop1 { get; set; }
  public string? Prop2 { get; set; }

  private int Init
  {
    [MemberNotNull(nameof(Prop1))] get => (Prop1 ??= """").GetHashCode();
    [MemberNotNull(nameof(Prop2))] set => Prop2 = value.ToString();
  }

  void M1()
  {
    Init++;
    Prop1.ToString();
    Prop2.ToString(); // warning
  }

  void M2()
  {
    Init = Init + 1;
    Prop1.ToString();
    Prop2.ToString();
  }
}
```

**Expected Behavior**:
No warnings. Both methods contain calls to both `get_Init()` and `set_Init()` before `Prop1` and `Prop2` are accessed

**Actual Behavior**:
Warning `CS8602: Dereference of a possibly null reference.` is reported for `Prop2.ToString()` in `M1` but not in `M2`
